packages/tools/devtools/devtools-view/src/components/TelemetryView.tsx

```diff
@@ -262,11 +313,15 @@ function FilteredTelemetryView(props: FilteredTelemetryViewProps): React.ReactEl
 		]);
 		// Initially matching options are all options
 		setMatchingOptions(eventNameOptionsRef.current);
-
 		const filtered = getFilteredEvents();
 		setFilteredTelemetryEvents(filtered);
+		if (filtered !== undefined) {
+			setIndex(index + (telemetryEvents.length - filtered?.length));
```

Comment:
You're observing the value of `index` here, so you will need to include it as a dependency of the Effect.